### DISCUSSION TITLE
How to deal with QuoteNone?

### DISCUSSION THREAD

**Discussion by dtslvr** [gadicc/yahoo-finance2#416] [Q&A] [answered]
I have upgraded to `yahoo-finance2@2.1.7` and have the following type error: `Property 'currency' does not exist on type 'QuoteNone'` for this snippet:

```
const quotes = await yahooFinance2.quote(['TSLA']);

for (const quote of quotes) {
  console.log('currency', quote.currency);
}
```

How can I deal with `QuoteNone`?

**Comment by gadicc** [comment]
Ugh.  I feel like we could have handled this better.  I'm going to look into this again, but in the meantime, you can do this:

```
  if (quote.quoteType !== "NONE")
    console.log('currency', quote.currency);
```

Tracking in https://github.com/gadicc/node-yahoo-finance2/issues/417.

**Reply by gadicc** [reply]
If you saw this and acted on it before I did the release, you'll need to remove the `if` line when upgrading :)  TypeScript will tell you as much.

**Comment by gadicc** [comment, marked as answer]
Ok, better answer... upgrade to v2.1.8 :)

Ref: #417

**Reply by dtslvr** [reply]
> Ok, better answer... upgrade to v2.1.8 :)

Awesome! That works like a charm 👍🏻
